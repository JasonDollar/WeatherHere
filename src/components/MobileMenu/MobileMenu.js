import React from 'react'
import PropTypes from 'prop-types'
import Icon from '../Icon/Icon'

import classes from './MobileMenu.module.scss'

const MobileMenu = ({
  text, showForecastHandler, showSearchHandler, showSettingsHandler, activeMenuClass,
}) => (
  <div className={classes.MobileMenu}>
    <button
      className={activeMenuClass === 'search' ? `${classes.active} ${classes.setting}` : classes.setting} 
      onClick={showSearchHandler}
    >
      <span className="hidden">{text.search}</span>
      <Icon icon="search" viewBox="0 0 50 50" title={text.icon} />
      <span>{text.search}</span>
    </button>
    <button 
      className={activeMenuClass === 'forecast' ? `${classes.active} ${classes.setting}` : classes.setting} 
      onClick={showForecastHandler}
    >
      <span className="hidden">{text.forecast}</span>
      <Icon icon="sun" viewBox="0 0 50 50" title={text.icon} />
      <span>{text.forecast}</span>
    </button>
    <button
      className={activeMenuClass === 'settings' ? `${classes.active} ${classes.setting}` : classes.setting} 
      onClick={showSettingsHandler}
    >
      <span className="hidden">{text.setings}</span>
      <Icon icon="settings" viewBox="0 0 50 50" title={text.icon} />
      <span>{text.settings}</span>
    </button>
  </div>
)

export default MobileMenu

MobileMenu.propTypes = {
  text: PropTypes.shape({
    search: PropTypes.string,
    settings: PropTypes.string,
    forecast: PropTypes.string,
    geo: PropTypes.string,
    geoFull: PropTypes.string,
    inputPlaceholder: PropTypes.string,
    icon: PropTypes.string,
  }).isRequired,
  showForecastHandler: PropTypes.func.isRequired, 
  showSearchHandler: PropTypes.func.isRequired, 
  showSettingsHandler: PropTypes.func.isRequired, 
  activeMenuClass: PropTypes.string.isRequired,
}