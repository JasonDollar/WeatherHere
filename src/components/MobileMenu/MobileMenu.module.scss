.MobileMenu {
  position: fixed;
  // top: calc(var(--screen-height) - 54px);
  height: 5.4rem;
  bottom: 0;
  left: 0;
  right: 0;
  // width: var(--screen-width);
  
  z-index: 250;
  
  display: flex;
  border-top: 1px solid var(--color-border);
  // box-shadow: 0 0 0 2px yellow;

  @media (min-resolution: 2dppx) {
    border-top: .5px solid var(--color-border);
    
  }
  @media (min-resolution: 3dppx) {
    border-top: .33333px solid var(--color-border);
    
  }
  @media (min-resolution: 4dppx) {
    border-top: .25px solid var(--color-border);
    
  }
  & .setting {
    color: var(--color-text-grey);
    background-color: var(--color-back-mobile);
    height: 100%;
    border: none;
    flex: 1;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    cursor: pointer;

    & p {
      font-size: var(--small-text-font-size);
      margin-bottom: .5rem;
    }

    & svg {
      height: var(--normal-text-font-size);
      width: var(--normal-text-font-size);
      fill: var(--color-text-grey);
      transform: scale(1.5);
      flex: 1
    }

    &.active {
      // background-color: var(--color-primary);
      color: var(--color-mobile-menu);
      & svg {
        fill: var(--color-mobile-menu);
      }
    }

  }
}