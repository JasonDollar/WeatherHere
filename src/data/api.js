const DARK_API = 'a077c3a2c30623e41ef4a1cafbec622d'
const DARK_URL = 'https://api.darksky.net/forecast/'

export { DARK_API, DARK_URL }  

