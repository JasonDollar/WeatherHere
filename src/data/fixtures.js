import weatherData from './random'
import languageText from './text-locale' 

export const appTestData = {
  weather: weatherData,
  units: 'si',
  language: 'pl',
  text: languageText,
  timezone: 'Europe/Warsaw',
  locationShortName: 'Plock',
  
}