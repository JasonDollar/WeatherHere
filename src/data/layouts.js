const layouts = {
  pink: {
    id: 'pink',
    name: 'Pink',
    primary: '#dd0055',
    primaryOffset: '#f5689e',
    mobileMenu: '#dd0055',
    antiGraph: '#8884d8',
    secondaryGraph: '#0c1052',
    background: '#fffcfc',
    background2: '#fffcfc',
    backgroundMobile: '#ddd',
    text: '#222',
    textSecondary: '#eee',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#bbb',
    grey3: '#eae9e9',
    greyGraph: '#676767',
  },
  pinkDark: {
    id: 'pinkDark',
    name: 'Pink Dark',
    primary: '#dd0055',
    primaryOffset: '#dd0055',
    mobileMenu: '#f5689e',
    antiGraph: '#8884d8',
    secondaryGraph: '#dd0055',
    background: '#272424',
    background2: '#fcfcfc',
    backgroundMobile: '#333',
    text: '#efeee7',
    textSecondary: '#ddd',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#999',
    grey3: '#333',
    greyGraph: '#acacac',
  },
  pumpkin: {
    id: 'pumpkin',
    name: 'Pumpkin Dark',
    primary: '#f48042',
    primaryOffset: '#ef915f',
    mobileMenu: '#f48042',
    antiGraph: '#50c25f',
    secondaryGraph: '#f48042',
    background: '#262424',
    background2: '#fcfcfc',
    backgroundMobile: '#333',
    text: '#eee',
    textSecondary: '#ddd',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#999',
    grey3: '#363333',
    greyGraph: '#acacac',
  },
  evening: {
    id: 'evening',
    name: 'Evening Dark',
    primary: '#b92b27',
    primaryOffset: '#d84f4b',
    mobileMenu: '#d84f4b',
    antiGraph: '#1565C0',
    secondaryGraph: '#d84f4b',
    background: '#282424',
    background2: '#fcfcfc',
    backgroundMobile: '#333',
    text: '#eeeef8',
    textSecondary: '#ddd',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#999',
    grey3: '#363333',
    greyGraph: '#acacac',
  },
  sky: {
    id: 'sky',
    name: 'Cool Sky',
    primary: '#2980B9',
    primaryOffset: '#6DD5FA',
    mobileMenu: '#2980B9',
    antiGraph: '#756ef0',
    secondaryGraph: '#2980B9',
    background: '#fcfcfc',
    background2: '#fcfcfc',
    backgroundMobile: '#ddd',
    text: '#222',
    textSecondary: '#eee',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#bbb',
    grey3: '#e7e7ec',
    greyGraph: '#676767',
  },
  harvey: {
    id: 'harvey',
    name: 'Harvey',
    primary: '#2b574b',
    primaryOffset: '#75d1a6',
    mobileMenu: '#386e60',
    antiGraph: '#0daeca',
    secondaryGraph: '#2b574b',
    background: '#fcfcfc',
    background2: '#fcfcfc',
    backgroundMobile: '#ddd',
    text: '#222',
    textSecondary: '#eee',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#bbb',
    grey3: '#e7eae7',
    greyGraph: '#676767',
  },


}

export default layouts



