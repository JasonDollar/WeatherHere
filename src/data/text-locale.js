const pl = {
  header: {
    title: 'Pogoda',
    search: 'Szukaj',
  },  
  layout: {
    search: 'Szukaj',
  },
  current: {
    today: 'Dziś',
    humidity: 'Wilgotność',
    wind: 'Wiatr',
    precipProb: 'Opady',
    precipType: 'Rodzaj opadu',
    precipInt: 'Intensywność opadu',
    appTemp: 'Temperatura odczuwalna',
    dewPoint: 'Punkt rosy',
    pressure: 'Ciśnienie',
    cloudCov: 'Zachmurzenie',
    visibility: 'Widoczność',
    uvIndex: 'Indeks UV',
    sunrise: 'Wschód słońca',
    sunset: 'Zachód słońca',
    moonphase: 'Faza księżyca',
    ozone: 'Ozon',
    windGust: 'Wiatr w porywach',
    location: 'Nieznana lokacja',
  },
  date: {
    weekDay: ['Niedziela', 'Poniedziałek', 'Wtorek', 'Środa', 'Czwartek', 'Piątek', 'Sobota', 'Dziś'],
    weekDayShort: ['Niedz.', 'Pon.', 'Wt.', 'Śr.', 'Czw.', 'Pt.', 'Sob.'],
    today: 'Dziś',
    temperature: 'Temperatura (°C)',
    temperatureMax: 'Temperatura Max.',
    temperatureMin: 'Temperatura Min.',
  },
  hourly: {
    hour: 'Godzina',
    precipProb: 'Możliwość Opadów (%)',
    precipProbShort: 'Możliwość Opadów',
    temperature: 'Temperatura (°C)',
    temperatureShort: 'Temperatura',
  },
  update: 'Ostatnia aktualizacja: ',
}

const en = {
  header: {
    title: 'Weather',
    
  },
  layout: {
    search: 'Search',
  },
  current: {
    today: 'Today',
    humidity: 'Humidity',
    wind: 'Wind',
    precipProb: 'Precipitation chance',
    precipType: 'Precipitation type',
    precipInt: 'Precipitation intensity',
    appTemp: 'Apparent temperature',
    dewPoint: 'Dew point',
    pressure: 'Pressure',
    cloudCov: 'Cloud cover',
    visibility: 'Visibility',
    uvIndex: 'UV Index',
    sunrise: 'Sunrise',
    sunset: 'Sunset',
    moonphase: 'Moon phase',
    ozone: 'Ozone',
    windGust: 'Wind gust',
    location: 'Unknown location',
  },
  date: {
    weekDay: ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'],
    weekDayShort: ['Sun.', 'Mon.', 'Tue.', 'Wed.', 'Thu.', 'Fri.', 'Sat.'],
    today: 'Today',
    temperature: 'Temperature (°C)',
    temperatureMax: 'Temperature Max.',
    temperatureMin: 'Temperature Min',
  },
  hourly: {
    hour: 'Hour',
    precipProb: 'Precipitation chance (%)',
    precipProbShort: 'Precipitation chance',
    temperature: 'Temperature (°C)',
    temperatureShort: 'Temperature',
  },
  update: 'Last updated: ',
}

export { pl, en } 