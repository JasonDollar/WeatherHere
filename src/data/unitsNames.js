export const unitsNames = {
  si: {
    id: 'si',
    distance: 'km',
    precip: 'mm/h',
    precipAcc: 'cm',
    temperature: '째C',
    wind: 'm/s',
    pressure: 'hPa',
  },
  us: {
    id: 'us',
    distance: 'mi',
    precip: 'in/h',
    precipAcc: 'in',
    temperature: '째F',
    wind: 'mi/h',
    pressure: 'hPa',
  },
  uk2: {
    id: 'uk2',
    distance: 'mi',
    precip: 'mm/h',
    precipAcc: 'cm',
    temperature: '째C',
    wind: 'mi/h',
    pressure: 'hPa',
  },
  ca: {
    id: 'ca',
    distance: 'km',
    precip: 'mm/h',
    precipAcc: 'cm',
    temperature: '째C',
    wind: 'km/h',
    pressure: 'hPa',
  },
}