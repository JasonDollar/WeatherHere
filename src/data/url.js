const DARK_URL = 'https://api.darksky.net/forecast/'

export { DARK_URL }