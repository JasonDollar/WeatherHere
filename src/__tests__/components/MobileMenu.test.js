import React from 'react'
import { shallow } from 'enzyme'
import { appTestData as data } from '../../data/fixtures'
import MobileMenu from '../../components/MobileMenu/MobileMenu'

const testProps = {
  text: data.text.pl.layout,
  showForecastHandler: jest.fn(),
  showSearchHandler: jest.fn(),
  showSettingsHandler: jest.fn(),
  activeMenuClass: 'forecast',
}
let wrapper

beforeEach(() => {
  wrapper = shallow(<MobileMenu {...testProps} />)
})

describe('MobileMeny test', () => {
  test('Matches snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('should handle search button click', () => {
    const element = wrapper.find('button').at(0)
    element.simulate('click')
    expect(testProps.showSearchHandler).toHaveBeenCalledTimes(1)
    // expect(element.hasClass('active')).toEqual(true)
  })

  test('should handle forecast button click', () => {
    const element = wrapper.find('button').at(1)
    element.simulate('click')
    expect(testProps.showForecastHandler).toHaveBeenCalledTimes(1)
    
  })

  test('should handle settings button click', () => {
    const element = wrapper.find('button').at(2)
    element.simulate('click')
    expect(testProps.showSettingsHandler).toHaveBeenCalledTimes(1)
    
  })
})