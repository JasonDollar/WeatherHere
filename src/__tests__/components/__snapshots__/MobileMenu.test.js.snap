// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileMenu test Matches snapshot 1`] = `
<div
  className="MobileMenu"
>
  <button
    className="setting"
    onClick={[MockFunction]}
    type="button"
  >
    <span
      className="hidden"
    >
      Szukaj
    </span>
    <Icon
      icon="search"
      title=""
      viewBox="0 0 50 50"
    />
    <span>
      Szukaj
    </span>
  </button>
  <button
    className="active setting"
    onClick={[MockFunction]}
    type="button"
  >
    <span
      className="hidden"
    >
      Pogoda
    </span>
    <Icon
      icon="sun"
      title=""
      viewBox="0 0 50 50"
    />
    <span>
      Pogoda
    </span>
  </button>
  <button
    className="setting"
    onClick={[MockFunction]}
    type="button"
  >
    <span
      className="hidden"
    />
    <Icon
      icon="settings"
      title=""
      viewBox="0 0 50 50"
    />
    <span>
      Ustawienia
    </span>
  </button>
</div>
`;
